periphery/src/Router.sol

```diff
@@ -105,29 +67,55 @@ contract Router {
         shares = lender.deposit(amount, msg.sender);
     }
 
-    function repayWithApprove(Lender lender, uint256 amount, address beneficiary) external returns (uint256 units) {
-        lender.asset().safeTransferFrom(msg.sender, address(lender), amount);
+    function repayWithPermit2(
+        Lender lender,
+        uint256 amount,
+        address beneficiary,
+        uint256 nonce,
+        uint256 deadline,
+        bytes calldata signature
```

Comment:
I might not have understood the `Permit2` docs correctly, so please correct me if I'm wrong. Clients of this function must call `ERC20(token).approve(Permit2)`. Is this `Permit2` Contract the same one that has been passed to the `Router`?